mergify_engine/backports.py

```diff
@@ -0,0 +1,196 @@
+                commit = commit.parents[0]
+            else:
+                # NOTE(sileht): What is that? A merge here?
+                LOG.error("%s: backport, unhandled commit structure",
+                          pull.pretty())
+                return []
+        return out_commits
+
+    elif len(commit.parents) == 2:
+        LOG.info("%s: backport, was just merged", pull.pretty())
+        # NOTE(sileht): Was merged, we can take commit from the PR
+        return list(filter(lambda c: not is_base_branch_merge_commit(c),
+                           sorted(pull.get_commits(), key=CommitOrderingKey)))
+    else:
+        # NOTE(sileht): What is that ?
+        LOG.error("%s: backport, unhandled commit structure", pull.pretty())
+        return []
+
+
+def _backport(repo, pull, branch_name, installation_token):
+    try:
+        branch = repo.get_branch(branch_name)
+    except github.UnknownObjectException:
+        LOG.info("%s doesn't exist for repo %s", branch_name, repo.full_name)
+        return
+
+    bp_branch = "mergify/bp/%s/pr-%s" % (branch_name, pull.number)
+
+    cherry_pick_fail = False
+    body = ("This is an automated backport of pull request #%d done "
+            "by Mergify.io" % pull.number)
+
+    repo.create_git_ref(ref="refs/heads/%s" % bp_branch, sha=branch.commit.sha)
+
+    git = utils.Gitter()
+
+    # TODO(sileht): This can be done with the Github API only I think:
+    # An example:
+    # https://github.com/shiqiyang-okta/ghpick/blob/master/ghpick/cherry.py
+    try:
+        git("clone", "--depth=1", "-b", bp_branch,
+            "https://x-access-token:%s@github.com/%s/" %
+            (installation_token, repo.full_name), ".")
+        git("config", "user.name", "%s-bot" % config.CONTEXT)
+        git("config", "user.email", config.GIT_EMAIL)
+
+        merge_commit = repo.get_commit(pull.merge_commit_sha)
+        for commit in _get_commits_to_cherrypick(pull, merge_commit):
+            # FIXME(sileht): Github does not allow to fetch only one commit
+            # So we have to fetch the branch since the commit date ...
+            # git("fetch", "origin", "%s:refs/remotes/origin/%s-commit" %
+            #    (commit.sha, commit.sha)
+            #    )
+            last_commit_date = commit.commit.committer.date
+            git("fetch", "origin", pull.base.ref,
+                "--shallow-since='%s'" % last_commit_date)
+            try:
+                git("cherry-pick", "-x", commit.sha)
+            except subprocess.CalledProcessError:
+                cherry_pick_fail = True
+                status = git("status").decode("utf8")
+                git("add", "*")
+                git("commit", "-a", "--no-edit")
+
+                body += ("\n\nCherry-pick of %s have failed:\n```\n%s```\n\n"
+                         % (commit.sha, status))
+
+        git("push", "origin", bp_branch)
+    except Exception:
+        LOG.exception("backport failure")
```

Comment:
not a fan of log.exception, rather log.<level> exc_info=True, you also lack contetxt info here I guess.